src/tx-builder.ts

```diff
@@ -271,16 +271,21 @@ export class TxBuilder {
 
         const { tmpdir, rmTmpDir } = this.prepareMoveBuild();
 
-        const { modules, dependencies, digest } = JSON.parse(
-            execSync(`sui move build --dump-bytecode-as-base64 --path ${path.join(moveDir, packageName)} --install-dir ${tmpdir}`, {
-                encoding: 'utf-8',
-                stdio: 'pipe', // silent the output
-            }),
-        );
-
-        rmTmpDir();
+        try {
+            const { modules, dependencies, digest } = JSON.parse(
+                execSync(`sui move build --dump-bytecode-as-base64 --path ${path.join(moveDir, packageName)} --install-dir ${tmpdir}`, {
+                    encoding: 'utf-8',
+                    stdio: 'pipe', // silent the output
+                }),
+            );
 
-        return { modules, dependencies, digest };
+            return { modules, dependencies, digest };
+        } catch (e) {
+            console.log(e);
```

Comment:
should be the caller's choice whether to log it or not (if the error isn't handled then it'll be shown to the user anyways